### PR TITLE
Null guard all interpolated strings passed to _t

### PR DESCRIPTION
**Pull request by Kegsay**
Comment should explain why.

### CODE REVIEW

File: src/languageHandler.js [hunk 1]
```diff
@@ -35,6 +35,26 @@ counterpart.setFallbackLocale('en');
...
 export function _t(...args) {
+    // Horrible hack to avoid https://github.com/vector-im/riot-web/issues/4191
+    // The interpolation library that counterpart uses does not support undefined/null
+    // values and instead will throw an error. This is a problem since everywhere else
+    // in JS land passing undefined/null will simply stringify instead, and when converting
+    // valid ES6 template strings to i18n strings it's extremely easy to pass undefined/null
+    // if there are no existing null guards. To avoid this making the app completely inoperable,
+    // we'll check all the values for undefined/null and stringify them here.
+    if (args[0]) {
+        const isInterpolation = ("" + args[0]).indexOf("%(") !== -1;
```
Comment (richvdh) [inline src/languageHandler.js]: 1. what's `"" + ` here for? why would `args[0]` be truthy but not a string already?
2. why are you even bothering with this? why not just `if (args[1] && typeof args[1] === 'object')` ?

**Comment by Kegsay** [comment]
> Is it our intention to keep this forever?

Yes.

>  It seems like a shame to have to do this check every single time we do a translation

It needs to be done either here or further up the stack in the actual code. Performance is hardly going to be an issue here.

> a better solution is not to pass in null/undefined in the first place

If you want to add null guards to all interpolated values knock yourself out, I think it's a waste of time. `null` or `undefined` values are completely valid in ES6 template strings so we've never had a reason to check for them in the code previously. As a result, it's likely that there will be fun edge cases where they can be `undefined` resulting in the app crashing. Alternative suggestions welcome.

File: src/languageHandler.js (hunk 1 above)
Comment (Kegsay) [inline src/languageHandler.js]: - `""+` will stringify whatever `args[0]` is.
- `args[0]` could be anything given it's truthy (an object, an array, a boolean) and only `string` has `indexOf` so it'll crash if you try to call that on anything else.
- Annoyingly `counterpart` changes its behaviour depending on what args you pass it. The `%(` check is an attempt to detect when we're using it for interpolating strings, so we don't needlessly touch `args[1]`.

Comment (richvdh) [inline src/languageHandler.js]: yeah, but do we ever actually use any of those alternative behaviours? Feels to me like the check for `%(` is costing more than it buys.

**Comment by richvdh** [comment]
> Performance is hardly going to be an issue here.

uhhh... debatable.

My only other suggestion (which apparently I forgot to write) would be to warn when we're making this replacement, so that we can try to fix them further up the stack.  I agree it's marginal though.

Also: tests have failed, though I suspect it's not your fault.

File: src/languageHandler.js (hunk 1 above)
Comment (Kegsay) [inline src/languageHandler.js]: I don't know if we use those alternative behaviours, as we call this function in a bazillion places so I can't reasonably check them.

**Comment by Kegsay** [comment]
> warn when we're making this replacement

Done. Is there anything else for me to do here? Are we keeping the `%(` check or removing it?

File: src/languageHandler.js (hunk 1 above)
Comment (richvdh) [inline src/languageHandler.js]: yeah, but what's the worst case? I can't imagine a case where `if (args[1] && typeof args[1] === 'object')` will be true, and replacing the null/undef members will be harmful.

**Comment by richvdh** [comment]
> Are we keeping the %( check or removing it?

I vote for removing it. Up to you at the end of the day, though.

LGTM otherwise.

**Comment by Kegsay** [comment]
Removed the check.

**Pull request merged**